src/SplQueue/LegacyImplementation.php

```diff
@@ -0,0 +1,90 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Laminas\Stdlib\SplQueue;
+
+use Serializable;
+use SplQueue;
+use UnexpectedValueException;
+
+use function get_class;
+use function gettype;
+use function is_array;
+use function is_object;
+use function serialize;
+use function sprintf;
+use function unserialize;
+
+/**
+ * Serializable version of SplQueue
+ */
+class LegacyImplementation extends SplQueue implements Serializable
```

Comment:
Should be `@internal` IMO: the implementation should not be part of the public API